src/StakeRegistry.sol

```diff
@@ -283,71 +251,111 @@ contract StakeRegistry is VoteWeigherBase, StakeRegistryStorage {
+        uint256 historyLength = operatorIdToStakeHistory[operatorId][quorumNumber].length;
+        if (historyLength != 0) {
+            operatorIdToStakeHistory[operatorId][quorumNumber][historyLength - 1]
                 .nextUpdateBlockNumber = uint32(block.number);
-            // load stake before update into memory if it exists
-            stakeBeforeUpdate = operatorIdToStakeHistory[operatorId][quorumNumber][operatorStakeHistoryLength - 1]
-                .stake;
+
+            prevStake = 
+                operatorIdToStakeHistory[operatorId][quorumNumber][historyLength - 1].stake;
         }
-        // push new stake to storage
-        operatorIdToStakeHistory[operatorId][quorumNumber].push(operatorStakeUpdate);
-        return stakeBeforeUpdate;
+        
+        // Create a new stake update and push it to storage
+        // TODO - update the entry instead of pushing, if the last update block is this block
+        operatorIdToStakeHistory[operatorId][quorumNumber].push(OperatorStakeUpdate({
+            updateBlockNumber: uint32(block.number),
+            nextUpdateBlockNumber: 0,
+            stake: newStake
+        }));
+
+        emit StakeUpdate(operatorId, quorumNumber, newStake);
+
+        // Return the change in stake
+        return _calculateDelta({ prev: prevStake, cur: newStake });
     }
 
     /// @notice Records that the `totalStake` for `quorumNumber` is now equal to the input param @_totalStake
-    function _recordTotalStakeUpdate(uint8 quorumNumber, OperatorStakeUpdate memory _totalStake) internal {
-        uint256 _totalStakeHistoryLength = _totalStakeHistory[quorumNumber].length;
-        if (_totalStakeHistoryLength != 0) {
-            _totalStakeHistory[quorumNumber][_totalStakeHistoryLength - 1].nextUpdateBlockNumber = uint32(block.number);
+    function _recordTotalStakeUpdate(uint8 quorumNumber, int256 stakeDelta) internal {
+        /**
+         * If this quorum has previous stake history, update the previous entry
+         * and fetch the previous total stake
+         */
+        uint96 prevStake;
+        uint256 historyLength = _totalStakeHistory[quorumNumber].length;
+        if (historyLength != 0) {
+            _totalStakeHistory[quorumNumber][historyLength - 1].nextUpdateBlockNumber = uint32(block.number);
+
+            prevStake = _totalStakeHistory[quorumNumber][historyLength - 1].stake;
+        }
+        
+        // Apply the stake delta to the previous stake, and push an update to the
+        // quorum's stake history
+        _totalStakeHistory[quorumNumber].push(OperatorStakeUpdate({
+            updateBlockNumber: uint32(block.number),
+            nextUpdateBlockNumber: 0,
+            stake: _applyDelta(prevStake, stakeDelta)
+        }));
+    }
+
+    /// @notice Returns the change between a previous and current value as a signed int
+    function _calculateDelta(uint96 prev, uint96 cur) internal pure returns (int256) {
+        if (cur >= prev) {
```

Comment:
`return int(cur) - int(prev)` should work?

Reply:
i knew there was a better way to do this! good catch.